packages/@aws-cdk/aws-ec2-alpha/lib/subnet-v2.ts

```diff
@@ -250,6 +252,23 @@ export class SubnetV2 extends Resource implements ISubnetV2 {
       subnet: this,
     });
   }
+
+  /**
+   * Return the Route Table associated with this subnet
+   */
+  public get routeTable(): IRouteTableV2 {
+    return this._routeTable;
+  }
+
+  /**
+   * Associate a Route Table with this subnet.
+   * @param routeTableProps The Route Table to associate with this subnet.
+   */
+  public associateRouteTable(routeTableProps: IRouteTableV2) {
+    this._routeTable = routeTableProps;
+    this.routeTableAssociation.addPropertyOverride('RouteTableId', routeTableProps.routeTableId);
```

Comment:
Currently we are doing an override rather than creating a new `CfnSubnetRouteTableAssociation` L1. If we create a new one, we would have to find some way to remove the old association L1, otherwise they will both show up in the template, and may confuse users who synthesize and see two associations for one table. The override allows us to use the same L1 object, just alter which route table it points to.